### PR TITLE
[Merged by Bors] - Change code to adhere to ADR18

### PR DESCRIPTION
**Pull request by soenkeliebau**
# Description
This contains a breaking change, since the version that is specified for Airflow now has to include a stackable image version.

Previously the version could be specified as "version: 3.2.1" and the operator simply added the stackable version to this. But going forward we have decided that the image version has to be explictly specified, so it will now have to be "version: 3.2.1-stackable-0.1.0" or similar.

This is in accordance with ADR18 (https://docs.stackable.tech/home/contributor/adr/ADR018-product_image_versioning.html).

*Please add a description here. This will become the commit message of the merge request later.*

## Review Checklist

- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/operator-binary/src/airflow_db_controller.rs [hunk 1]
```diff
@@ -188,7 +188,7 @@ fn build_init_job(airflow_db: &AirflowDB) -> Result<Job> {
 
     let container = ContainerBuilder::new("airflow-init-db")
         .image(format!(
-            "docker.stackable.tech/stackable/airflow:{}-stackable0",
```
Comment (adwk67) [inline rust/operator-binary/src/airflow_db_controller.rs:191]: One hard-coded reference to `stackable0` still exists in `airflow_controller.rs` that should be removed in a similar way:
https://github.com/stackabletech/airflow-operator/blob/feat/adr18/rust/operator-binary/src/airflow_controller.rs#L288

File: examples/simple-airflow-cluster.yaml [hunk 2]
```diff
@@ -20,7 +20,7 @@ kind: AirflowCluster
 metadata:
   name: airflow
 spec:
-  version: 2.2.3-python38
+  version: 2.2.3-python38-stackable0.3.0
```
Comment (adwk67) [inline examples/simple-airflow-cluster.yaml:23]: This change is missing in the other two examples.

File: rust/operator-binary/src/airflow_db_controller.rs (hunk 1 above)
Comment (soenkeliebau) [reply rust/operator-binary/src/airflow_db_controller.rs:191]: Removed

File: examples/simple-airflow-cluster.yaml (hunk 2 above)
Comment (soenkeliebau) [reply examples/simple-airflow-cluster.yaml:23]: fixed

**Review by adwk67 — APPROVED** [review]
LGTM.

**Pull request closed without merging**